.github/workflows/ci-core.yml

```diff
@@ -518,15 +526,41 @@ jobs:
         run: |
           make rm-mocked
           make generate
-      - name: Ensure clean after generate
-        run: |
-          git add --all
-          git diff --stat --cached --exit-code
-      - run: make gomodtidy
-      - name: Ensure clean after tidy
+
+      # If asked to fix issues, set up Git user and GitHub token to make the commit.
+      - name: Set up Git user to commit fixes
+        if: ${{ always() && contains( github.event.pull_request.labels.*.name, 'auto-fix') }}
+        shell: bash
         run: |
-          git add --all
-          git diff --minimal --cached --exit-code
```

Comment:
This will no longer fail if changes exist, regardless of the existence of the `auto-fix` label.